### DISCUSSION TITLE
Cypress - webpack compilation error during the pageobjects import in a test case

### DISCUSSION THREAD

**Discussion by faprile** [cypress-io/cypress#15019] [Component Testing]
Hello everyone,
I'm trying to set up my first pageobject in cypress but I had this error:

`Error: Webpack Compilation Error
./cypress/integration/test/login2.spec.js
Module not found: Error: Can't resolve 'integration/pageObjects/login' in 'C:\Users\USER\Documents\cypress\cypress\integration\test'
resolve 'integration/pageObjects/login' in 'C:\Users\USER\Documents\cypress\cypress\integration\test'
  Parsed request is a module
  using description file: C:\Users\USER\Documents\cypress\cypress\package.json (relative path: ./integration/test)
    Field 'browser' doesn't contain a valid alias configuration`

my test:

`import Login from 'integration/pageObjects/login'

describe('Login', () => {
  const login = new Login()
    it('Sign in', () => {
      cy.fixture("data").then((data) => {

        const user = data.username;
        const pass = data.password;
        const url = data.baseUrl;

      cy.visit(url)
  
      login.name().type(user) 
  
      login.pw().type(pass)   
  
      login.loginbutton().click()  
      
      var bar
      bar = cy.get('.GJ4LAWHCBGC.GJ4LAWHCEL.x-ignore.GJ4LAWHCI2E', { timeout: 10000 }).should('be.visible')
      bar.eq(4).click()
      
      cy.contains('Logout').click()
    })
  })
})`

and my page object: 

`class Login {

    name(){
        return cy.get('input[placeholder="User Name"]', { timeout: 10000 }).should('be.visible')
    }

    pw() {
        return cy.get('input[placeholder="Password"]', { timeout: 10000 }).should('be.visible')
    }

    loginbutton() {
        return cy.contains('Login', { timeout: 10000 }).should('be.visible') 
    }

}

export default Login`

What could be the problem?
Thanks to all!

**Comment by bahmutov** [comment]
Is the `import Login from 'integration/pageObjects/login'` correct path from the spec file? It should be a relative path, right?

**Reply by faprile** [reply]
I thought that could be this the problem, but I don't know how to set up it in the spec file

**Reply by faprile** [reply]
in a matter of fact i received this message: module "c:/Users/USER/Documents/cypress/cypress/integration/pageObjects/login"
'login' is declared but its value is never read.ts(6133)

**Reply by faprile** [reply]
" > Login is not defined" 
const login = new Login()

**Reply by faprile** [reply]
import: solved. I changed the path with the correct path.
new Login(): solved. I changed and correct the problem in "const lg = new login()"
now it works. 
Thanks you very much!
